src/vaadin-combo-box-data-provider-mixin.js

```diff
@@ -169,6 +169,12 @@ export const ComboBoxDataProviderMixin = (superClass) =>
       if (!this._pendingRequests[page] && this.dataProvider) {
         this.loading = true;
 
+        // do not clear filter if auto open is disabled
+        // otherwise, the first page loaded will not be filtered
+        if (this.autoOpenDisabled) {
+          this.filter = this.filter || this._inputElementValue;
+        }
```

Comment:
I would suggest to go with a slightly different approach to avoid extra `clearCache` call. 

So, here is a problem:

1. When combo-box has `autoOpenDisabled`, and user types a value, both `value` and `filter` are set.
2. The following observer is called because `value` is set before opening the combo-box:
https://github.com/vaadin/vaadin-combo-box/blob/f7fdbe508adfca584216cf0578c922fbfc8d39c5/src/vaadin-combo-box-data-provider-mixin.js#L81-L86
3. `this.filter` check returns `true` and logic inside `id` is executed, so it resets the `filter`, which is a bug.

Proposed fix: update the observer condition to look like this:

```js
if (dataProvider && !this.loading && this.filter && !(this.autoOpenDisabled && value === this.filter)) {
```

That's exactly what happens: `value` and `filter` are the same, and `autoOpenDisabled` is true.